internal/painter/font.go

```diff
@@ -313,7 +326,8 @@ type FontCacheItem struct {
 	Fonts shaping.Fontmap
 }
 
-var fontCache = &sync.Map{} // map[fyne.TextStyle]*FontCacheItem
+var fontCache = &sync.Map{}       // map[fyne.TextStyle]*FontCacheItem
+var fontCustomCache = &sync.Map{} // map[string]*FontCacheItem for custom resources
```

Comment:
I think we are missing a clean task for fontCustomCache

Reply:
It should be there - line 154

Reply:
OK, yeah I see it, thanks. The possible issue is- that particular clean task only runs on settings or theme change. Which was fine when we could only load a limited number of fonts because it was only what the theme supports. And even with OS font lookup we technically have a cap on the number of fonts that will be loaded because it's at most one per Unicode script.

But with this API the app can load an unlimited number of fonts so it is more of a true memory leak now since it's unbounded. Not sure if it would be blocking but at least we will need to file a follow-up ticket to address this sooner than later probably.

Reply:
I don't know exactly how or when we should clean out fonts that have been used in the app but not used for a while. A follow up sounds reasonable.

Reply:
#4908

Reply:
thanks